### DISCUSSION TITLE
[http-connection] - static: Client connection failed with error 30 (AWS_ERROR_PRIORITY_QUEUE_EMPTY)

### DISCUSSION THREAD

**Discussion by ramkumarpvs** [aws/aws-iot-device-sdk-cpp-v2#611] [Q&A] [answered]
With latest aws-iot-device-sdk-cpp-v2 latest code and FIPS enabled crypto we are seeing that connection(client TLS) fails with following error
**[http-connection] - static: Client connection failed with error 30 (AWS_ERROR_PRIORITY_QUEUE_EMPTY)..** 
Eventually our application mqtt->connect fails with this error:
error aws-c-auth: AWS_AUTH_SIGNING_NO_CREDENTIALS, Attempt to sign an http request without credentials.

[TRACE] [2023-08-08T17:35:38Z] [00007fdbba073700] [channel-bootstrap] - id=0x5623e33d1210: Setting up socket handler on channel 0x7fdbb00018d0 with handler 0x7fdbb0012890 on slot 0x7fdbb00122e0.
[TRACE] [2023-08-08T17:35:38Z] [00007fdbba073700] [channel] - id=0x7fdbb00018d0: creating new slot 0x7fdbb0000fc0.
[ERROR] [2023-08-08T17:35:38Z] [00007fdbba073700] **[channel-bootstrap] - id=0x5623e33d1210: channel 0x7fdbb00018d0 setup failed with error 30.**
[TRACE] [2023-08-08T17:35:38Z] [00007fdbba073700] [channel] - id=0x7fdbb00018d0: channel shutdown task is scheduled

Attaching the trace and error logs.
[aws-iot-device-sdk-cpp-v2-errors.txt](https://github.com/aws/aws-iot-device-sdk-cpp-v2/files/12294470/aws-iot-device-sdk-cpp-v2-errors.txt)

**Comment by bretambrose** [comment, marked as answer]
The error code is a red herring.  You've hit a code path that fails but does not set the thread-local error code and so whatever was set previously is being surfaced.

Based on the logging, I believe the failing call is here:

https://github.com/awslabs/aws-c-io/blob/main/source/channel_bootstrap.c#L299-L300

Without knowing more about platform, compilation setup, and what "FIPS enabled crypto" implies, I can't say more.

I would suggest stepping into the function above and seeing what exactly is failing.
